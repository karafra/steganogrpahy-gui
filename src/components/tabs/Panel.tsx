import React, { ReactNode } from "react";

export interface PanelProps {
  title: string;
  children: ReactNode[] | ReactNode;
  width?: string; 
}

const Panel = (props: PanelProps) => {
  return <div style={{ alignContent: "center", justifyContent: "center" }}>{props.children}</div>;
};

export default Panel;
