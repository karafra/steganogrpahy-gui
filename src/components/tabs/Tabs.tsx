import React, { ReactElement, useState } from 'react';
import { PanelProps } from './Panel';
import styles from "components/tabs/scss/Tabs.module.scss";

export interface TabsProps {
  selected?: number;
  children: ReactElement<PanelProps>[];
  width?: string;
}

const Tabs = (props: TabsProps) => {
  const [selected, setSelected] = useState(props.selected || 0);
  const handleChange = (index: number) => {
      setSelected(index);
  };
  return (
    <div className={styles.container}>
      <ul className={styles.inline}>
        {props.children.map((elem, index)=>{
          let style = index === selected ? styles.selected: undefined;
         return <li className={style} style={{
           width: props.width,
        }} key={index} onClick={handleChange.bind(this,index)}>{elem.props.title}</li>
        })}
      </ul>
      <div className={styles.tab}>{props.children[selected]}</div>
      </div>
    );
}

export default Tabs;
