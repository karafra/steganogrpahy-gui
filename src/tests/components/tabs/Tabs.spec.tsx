import React from "react";
import renderer, { act } from "react-test-renderer";
import Panel from "../../../components/tabs/Panel";
import Tabs from "../../../components/tabs/Tabs";
import { shallow, mount } from "enzyme";

describe("Tabs tests", () => {
  const testTitle = "test title";
  const testChildren = "test children";
  describe("snapshot tests", () => {
    test("should render component", () => {
      // Given
      const component = renderer.create(
        <Tabs selected={0}>
          <Panel title={testTitle}>{testChildren}</Panel>
          <Panel title={testTitle}>{testChildren}</Panel>
        </Tabs>
      );
      // When
      const tree = component.toJSON();
      // Then
      expect(tree).toMatchSnapshot();
    });
  });
});
