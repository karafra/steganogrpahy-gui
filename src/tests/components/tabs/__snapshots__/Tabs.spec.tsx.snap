// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs tests snapshot tests should render component 1`] = `
<div
  className="container"
>
  <ul
    className="inline"
  >
    <li
      className="selected"
      onClick={[Function]}
      style={
        Object {
          "width": undefined,
        }
      }
    >
      test title
    </li>
    <li
      onClick={[Function]}
      style={
        Object {
          "width": undefined,
        }
      }
    >
      test title
    </li>
  </ul>
  <div
    className="tab"
  >
    <div
      style={
        Object {
          "alignContent": "center",
          "justifyContent": "center",
        }
      }
    >
      test children
    </div>
  </div>
</div>
`;
